### DISCUSSION TITLE
Reclustering cell-types and differential expression after SCTransform and Integration

### DISCUSSION THREAD

**Discussion by srira25** [satijalab/seurat#4076] [Q&A]
Hello @satijalab ,

I have been working with Seyrat v4 for a large dataset with 18 samples and 2 different conditions(9 per condition). I have integrated the data into a large Seurat object using the pipeline provided on the Seurat website and clustered them to identify 7 cell types. Now, I want to individually extract the cells from each cell-type and do reclustering to identify sub-types and differential expression for each individually.

1. My workflow is as below where the SCT assay of Seurat.integrated has been used for differential expression:
               DefaultAssay(Seurat.integrated) <- "SCT"
               OverallDiffexp <- FindMarkers(Seurat.integrated, ident.1 = "CONTROL", ident.2 = "STIM", verbose = TRUE)

2. Subset the dataset, and identify individual markers for CONTROL vs STIM using the SCT assay for each cell type separately:
Seurat.cell_types <- lapply(cell_types, function(x) {
  y <- subset(Seurat.integrated, idents = x)
  return(y)
})
diff.response <- lapply(Seurat.cell_types, function(x){
  DefaultAssay(x) <- "SCT"
  y <- FindMarkers(x, ident.1 = "CONTROL", ident.2 = "STIM", verbose = TRUE)
  return(y)
})

3. Re-cluster each cell type individually using the SCT pipeline: 
Seurat.cell_types_subclust <- lapply(Seurat.cell_types, function(x){
  DefaultAssay(x) <- "SCT"
  x <- FindVariableFeatures(object = x,selection.method = 'vst', nfeatures = 2000)
  x <- RunPCA(x, verbose = TRUE)
  x <- RunUMAP(x, dims = 1:30)
  x <- FindNeighbors(object = x, dims = 1:20)
  x <- FindClusters(object = x, resolution = c(0.4, 0.6, 0.8, 1.0, 1.4))
  return(x)
})

My questions were:
1. whether the SCT assay can be used for differential expression like this or is it better to use RNA assay and use NormalizeData() and ScaleData() on it before using FindMarkers for differential expression.
2. When I extract the individual cell types individually, do I need to start from the unintegrated data for all 18 samples and do the entire integration pipeline again, before proceeding to reclustering and differential expression, or is it fine to keep working on the integrated dataset. Are there any guidelines for this?
3. Is FindVariableFeatures necessary to do everytime I need to cluster the cells? Is there any specific pipeline I could adopt to do this?
4. How do I select the optimum resolution of FindClusters. Is there any guideline for this?

Any clarification would be great.

**Comment by jenriquezl337** [comment]
Hi @srira25, I'm just a fellow Seurat user and I have a similar question. It would be helpful for @satijalab to best answer, however I'll try to start the discussion with what I've done.

Question 1: Discussion #4032 is the most recent answer. I understand that the "RNA" assay is best for running DE after integration

Question 2: I don't think you need to start from the unintegrated data. This is where I get fuzzy and am also unsure. I usually run two different pipelines to first check how well integration is maintained: 
A)
subset <- subset(IntegratedObj,  idents =  c("A", "B"))
subset <- SCTransform(subset)
subset <- RunPCA(subset)
subset <- RunUMAP(subset, dims = 1:30)
subset  <- FindNeighbors(subset)
subset <-  FindClusters(subset)
 
or
 
B)
subset <- subset(OriginalObj,  idents =  c("A", "B"))
DefaultAssay(subset) <- 'integrated'
subset <- RunPCA(subset,  assay = 'integrated')
subset <- RunUMAP(subset, dims = 1:30, assay = 'integrated')
subset <- FindNeighbors(subset, assay = "integrated")
subset <- FindClusters(subset)

I have no idea which is the correct approach and hopefully we get clarification.

Question 3: I think it depends on the answer we get on how to recluster after integration. If running my option A, you don't need to run FindVariableFeatures because SCT has it built into the function. If running my option B, then you don't need to run it either. 

Question 4: Resolution is always tricky. I use package clustree to help me inform optimal resolution. 
https://cran.r-project.org/web/packages/clustree/vignettes/clustree.html

Hope this helps; again I think the Seurat team needs to provide clarification
